src/rngs/adapter/reseeding.rs

```diff
@@ -310,8 +316,16 @@ mod fork {
 
     pub fn register_fork_handler() {
         static REGISTER: Once = Once::new();
-        REGISTER.call_once(|| unsafe {
-            libc::pthread_atfork(None, None, Some(fork_handler));
+        REGISTER.call_once(|| {
+            // Bump the counter before and after forking (see #1169):
+            let ret = unsafe { libc::pthread_atfork(
+                Some(fork_handler),
+                Some(fork_handler),
+                Some(fork_handler),
+            ) };
+            if ret != 0 {
+                panic!("libc::pthread_atfork failed with code {}", ret);
```

Comment:
We might want to mention in the docs that this may panic. On the other hand, this can only fail due to an allocation failure, which is usually not documented in Rust functions.

Reply:
Does it change anything really? If it happens repeatedly on some platform then it's probably something we'll need to fix; otherwise it's just an obscure detail only of interest to people doing detailed code reviews.

Reply:
I think it's fine. If `pthread_atfork` fails, we will probably abort anyway.